### PR TITLE
Allow custom textbox background color

### PR DESCRIPTION
**Pull request by agaertner**
Small tweak to hide the background of a multiline textbox in order to use a custom one or to have it react dynamically on focus change.
The simple textbox uses a texture and I wasn't sure if I should redo it to be drawn by a single pixel like in the multiline textbox. I could remove the texture and allow color customization here as well.

### CODE REVIEW

**Comment by dlamkins** [comment]
Seems most intuitive / useful if we just have the option to hide the background.  Controls already have a background color, but if a custom background was needed it would probably be best to just throw the multiline textbox without a background into a container which draws what is needed.

**Comment by agaertner** [comment]
Like so?

File: Blish HUD/Controls/MultilineTextBox.cs [hunk 1]
```diff
@@ -8,6 +8,16 @@ public class MultilineTextBox : TextInputBase {
         private const int TEXT_TOPPADDING  = 7;
         private const int TEXT_LEFTPADDING = 10;
 
+        private bool _hideBackground;
+        public bool HideBackground {
+            get => _hideBackground;
+            set {
```
Comment (dlamkins) [inline Blish HUD/Controls/MultilineTextBox.cs]: Use `SetProperty` instead which automates all of this.

File: Blish HUD/Controls/MultilineTextBox.cs [hunk 2]
```diff
@@ -193,24 +203,26 @@ public class MultilineTextBox : TextInputBase {
         protected override void UpdateScrolling() { /* NOOP */ }
 
         protected override void Paint(SpriteBatch spriteBatch, Rectangle bounds) {
-            // Background tint
-            spriteBatch.DrawOnCtrl(this, ContentService.Textures.Pixel, new Rectangle(1, 1, bounds.Width - 2, bounds.Height - 2), Color.Black * 0.5f);
+            if (!HideBackground) {
```
Comment (dlamkins) [inline Blish HUD/Controls/MultilineTextBox.cs]: For consistency, let's do `this.HideBackground`.

File: Blish HUD/Controls/Textbox.cs [hunk 3]
```diff
@@ -26,6 +26,16 @@ public class TextBox : TextInputBase {
         /// </summary>
         public event EventHandler<EventArgs> EnterPressed;
 
+        private bool _hideBackground;
+        public bool HideBackground {
+            get => _hideBackground;
+            set {
```
Comment (dlamkins) [inline Blish HUD/Controls/Textbox.cs]: Use `SetProperty` instead which automates all of this.

File: Blish HUD/Controls/Textbox.cs [hunk 4]
```diff
@@ -129,14 +139,20 @@ public class TextBox : TextInputBase {
...
         protected override void Paint(SpriteBatch spriteBatch, Rectangle bounds) {
-            spriteBatch.DrawOnCtrl(this,
-                                   _textureTextbox,
-                                   new Rectangle(Point.Zero, _size - new Point(5, 0)),
-                                   new Rectangle(0, 0, Math.Min(_textureTextbox.Width - 5, _size.X - 5), _textureTextbox.Height));
-
-            spriteBatch.DrawOnCtrl(this, _textureTextbox,
-                                   new Rectangle(_size.X - 5, 0, 5, _size.Y),
-                                   new Rectangle(_textureTextbox.Width - 5, 0, 5, _textureTextbox.Height));
+            if (!HideBackground) {
```
Comment (dlamkins) [inline Blish HUD/Controls/Textbox.cs]: Same here, let's use `this.HideBackground`.

**Review by dlamkins — CHANGES_REQUESTED** [review]

**Comment by dlamkins** [comment]
Yeah, that's exactly what I meant.  🙂 Made a couple of notes for quick revisions that should only take a moment.  👍

**Comment by agaertner** [comment]
> Yeah, that's exactly what I meant. 🙂 Made a couple of notes for quick revisions that should only take a moment. 👍

All done 👍

**Review by dlamkins — APPROVED** [review]

**Comment by dlamkins** [comment]
Perfection.  Thanks!

**Pull request merged**